src/RecyclableMemoryStream.cs

```diff
@@ -1310,13 +1429,21 @@ private BlockAndOffset GetBlockAndRelativeOffset(long offset)
             return new BlockAndOffset(blockIndex, offsetIndex);
         }
 
-        private void EnsureCapacity(long newCapacity)
+        private void EnsureCapacity(long newCapacity, bool throwOomExceptionOnOverCapacity = false)
         {
             if (newCapacity > this.memoryManager.MaximumStreamCapacity && this.memoryManager.MaximumStreamCapacity > 0)
             {
                 this.memoryManager.ReportStreamOverCapacity(this.id, this.tag, newCapacity, this.AllocationStack);
-                throw new InvalidOperationException("Requested capacity is too large: " + newCapacity + ". Limit is " +
-                                                    this.memoryManager.MaximumStreamCapacity);
+                string message = $"Requested capacity is too large: {newCapacity}. Limit is {this.memoryManager.MaximumStreamCapacity}";
```

Comment:
Nitpicking: this has 2 boxing operations.

For 4 blocks, this is the best option:

```CSharp
string message = "Requested capacity is too large: " + newCapacity.ToString(CultureInfo.InvariantCulture) + ". Limit is " + this.memoryManager.MaximumStreamCapacity.ToString(CultureInfo.InvariantCulture);
```

But, in the general case, prefer:

```CSharp
string message = $"Requested capacity is too large: {newCapacity.ToString(CultureInfo.InvariantCulture)}. Limit is {this.memoryManager.MaximumStreamCapacity.ToString(CultureInfo.InvariantCulture)}";
```

Reply:
I don't expect this code path to be hit very much, but it's an easy fix.